File: core/src/main/java/javabot/web/views/ChangesView.java
package javabot.web.views;

import com.google.inject.Injector;
import javabot.dao.ChangeDao;
import javabot.dao.util.QueryParam;
import javabot.model.Change;

import javax.inject.Inject;
import javax.servlet.http.HttpServletRequest;
import java.util.List;

public class ChangesView extends PagedView<Change> {
    private final Change filter;
    @Inject
    private ChangeDao changeDao;

    public ChangesView(final Injector injector, final HttpServletRequest request, final int page, Change filter) {
        super(injector, request, page);
        this.filter = filter;
    }

    @Override
    public String getPagedView() {
        return "changes.ftl";
    }

    @Override
    public Change getFilter() {
        return filter;
    }

    @Override
    public Long countItems() {
        return changeDao.count(filter);
    }

    @Override
    protected String getPageUrl() {
        return "/changes";
    }

    @Override
    public List<Change> getPageItems() {
        return changeDao.getChanges(new QueryParam(getIndex(), ITEMS_PER_PAGE, "updated", true), filter);
    }
}

File: src/main/kotlin/javabot/web/views/ChangesView.kt
package javabot.web.views

import com.google.inject.assistedinject.Assisted
import jakarta.inject.Inject
import jakarta.servlet.http.HttpServletRequest
import java.time.LocalDateTime
import javabot.dao.AdminDao
import javabot.dao.ApiDao
import javabot.dao.ChangeDao
import javabot.dao.ChannelDao
import javabot.dao.FactoidDao
import javabot.dao.util.QueryParam
import javabot.model.Change
import javax.annotation.Nullable

class ChangesView
@Inject
constructor(
    adminDao: AdminDao,
    channelDao: ChannelDao,
    factoidDao: FactoidDao,
    apiDao: ApiDao,
    var changeDao: ChangeDao,
    @Assisted request: HttpServletRequest,
    @Assisted page: Int,
    @param:Nullable @param:Assisted private val message: String?,
    @param:Nullable @param:Assisted private val date: LocalDateTime?,
) : PagedView<Change>(adminDao, channelDao, factoidDao, apiDao, request, page) {

    override fun getPagedView(): String {
        return "changes.ftl"
    }

    override fun countItems(): Long {
        return changeDao.count(message, date)
    }

    override fun getPageUrl(): String {
        return "/changes"
    }

    override fun getPageItems(): List<Change> {
        return changeDao.getChanges(
            QueryParam(getIndex(), ITEMS_PER_PAGE, "updated"),
            message,
            date,
        )
    }
}
